src/Modal/Modal.tsx

```diff
@@ -0,0 +1,212 @@
+import * as React from 'react';
+import styled, { keyframes } from 'styled-components';
+import clsx from 'clsx';
+import Transition, { TransitionStatus } from 'react-transition-group/Transition';
+import ModalOverlay from './ModalOverlay';
+import { Box } from '../Box';
+import { Portal } from '../Portal';
+import { ANIMATION_DURATION } from './constants';
```

Comment:
Might be a good time to consider adding extra keys to the theme object - https://github.com/South-Paw/spicy-ui/projects/1#card-31840451

Then make use of `useTheme` to get the animation timings etc?